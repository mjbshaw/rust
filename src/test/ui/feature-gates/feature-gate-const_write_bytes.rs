// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![feature(const_fn, const_let)]

fn main() {}

const unsafe fn foo(u: *mut u32) {
    std::ptr::write_bytes(u, 0u8, 1);
    //~^ ERROR The use of std::ptr::write_bytes() is gated in constant functions (see issue #53491)
}
