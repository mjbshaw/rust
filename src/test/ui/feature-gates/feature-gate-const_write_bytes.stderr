error[E0658]: The use of std::ptr::write_bytes() is gated in constant functions (see issue #53491)
  --> $DIR/feature-gate-const_write_bytes.rs:16:5
   |
LL |     std::ptr::write_bytes(u, 0u8, 1);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(const_write_bytes)] to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
